src/Lecture1.hs

```diff
@@ -1,122 +1,137 @@
-difference between the biggest number and the smallest one.
-
->>> minmax 7 1 4
-6
-
-Explanation: @minmax 7 1 4@ returns 6 because 7 is the biggest number
-and 1 is the smallest, and 7 - 1 = 6.
-
-Try to use local variables (either let-in or where) to implement this
-function.
--}
-minmax x y z = error "TODO"
-
-{- | Implement a function that takes a string, start and end positions
-and returns a substring of a given string from the start position to
-the end (including).
-
->>> subString 3 7 "Hello, world!"
-"lo, w"
-
->>> subString 10 5 "Some very long String"
-""
-
-This function can accept negative start and end position. Negative
-start position can be considered as zero (e.g. substring from the
-first character) and negative end position should result in an empty
-string.
--}
-subString start end str = error "TODO"
-
-{- | Write a function that takes a String — space separated numbers,
-and finds a sum of the numbers inside this string.
-
->>> strSum "100    -42  15"
-73
-
-The string contains only spaces and/or numbers.
--}
-strSum str = error "TODO"
-
-{- | Write a function that takes a number and a list of numbers and
-returns a string, saying how many elements of the list are strictly
-greated than the given number and strictly lower.
-
->>> lowerAndGreater 3 [1 .. 9]
-"3 is greater than 2 elements and lower than 6 elements"
-
-Explanation: the list [1 .. 9] contains 9 elements: [1, 2, 3, 4, 5, 6, 7, 8, 9]
-The given number 3 is greater than 2 elements (1 and 2)
-and lower than 6 elements (4, 5, 6, 7, 8 and 9).
-
-🕯 HINT: Use recursion to implement this function.
--}
-lowerAndGreater n list = error "TODO"
+lastDigit :: Int -> Int
+lastDigit n = abs n `mod` 10
+
+-- | Write a function that takes three numbers and returns the
+-- difference between the biggest number and the smallest one.
+--
+-- >>> minmax 7 1 4
+-- 6
+--
+-- Explanation: @minmax 7 1 4@ returns 6 because 7 is the biggest number
+-- and 1 is the smallest, and 7 - 1 = 6.
+--
+-- Try to use local variables (either let-in or where) to implement this
+-- function.
+minmax :: Int -> Int -> Int -> Int
+minmax x y z =
+  let minNumber = min (min x y) z
+      maxNumber = max (max x y) z
+   in maxNumber - minNumber
+
+-- | Implement a function that takes a string, start and end positions
+-- and returns a substring of a given string from the start position to
+-- the end (including).
+--
+-- >>> subString 3 7 "Hello, world!"
+-- "lo, w"
+--
+-- >>> subString 10 5 "Some very long String"
+-- ""
+--
+-- This function can accept negative start and end position. Negative
+-- start position can be considered as zero (e.g. substring from the
+-- first character) and negative end position should result in an empty
+-- string.
+subString :: Int -> Int -> String -> String
+subString start end str
+  | end < 0 = ""
+  | otherwise = take ((end + 1) - posIndex start) (drop (posIndex start) str)
+  where
+    posIndex :: Int -> Int
+    posIndex n = if n < 0 then 0 else n
```

Comment:
You can write this function shorter using standard function `max` 🙂 
```suggestion
    posIndex n = max 0 n
```

Reply:
Thanks for review !